lingo-share/src/components/video/videochat.jsx

```diff
@@ -1,28 +1,44 @@
 import React, { useState, useCallback, useEffect } from "react";
 import { get_token } from "../../firebase";
+import { UserAuth } from "../../contexts/AuthContext";
+import { useNavigate } from "react-router-dom";
 import Video from "twilio-video";
 import Lobby from "./lobby";
 import Room from "./room";
 
-const VideoChat = () => {
+const VideoChat = ({callerID}) => {
   const [username, setUsername] = useState("");
   const [roomName, setRoomName] = useState("");
   const [room, setRoom] = useState(null);
   const [connecting, setConnecting] = useState(false);
+  const { user } = UserAuth();
+  const navigate = useNavigate();
 
-  const handleUsernameChange = useCallback((event) => {
-    setUsername(event.target.value);
+  const handleLogout = useCallback(() => {
+    setRoom((prevRoom) => {
+      if (prevRoom) {
+        prevRoom.localParticipant.tracks.forEach((trackPub) => {
+          trackPub.track.stop();
+        });
+        prevRoom.disconnect();
+      }
+      return null;
+    });
+    navigate("/dashboard");
   }, []);
 
-  const handleRoomNameChange = useCallback((event) => {
-    setRoomName(event.target.value);
-  }, []);
+  const getRoomName = (uid, callerID) => {
+    const concatenatedRoomName = uid + callerID;
```

Comment:
Let's change this to uid + callerID But sorted alphabetically, just make list and then call. .sort() and then combine